### PR TITLE
trilinos: add conflict and enable gtest for developers

### PR DESCRIPTION
**Pull request by sethrj**
Per discussion with @psakievich
- Add `+exodus~fortran` conflict  (before #25549 building `+exodus~fortran` would secretly enable Fortran)
- Enable vendored trilinos gtest package when `self.run_tests` for trilinos developers

I also:
- reordered some of the conflicts/deps in a separate commit to make them easier to find.
- removed the `zlib` variant in favor of always enabling GZIP compression in Zoltan when that package is enabled (there's such a thing as too-fine granularity in package options...)

### CODE REVIEW

File: var/spack/repos/builtin/packages/trilinos/package.py
Comment (psakievich) [inline var/spack/repos/builtin/packages/trilinos/package.py]:
```suggestion
            define_trilinos_enable('Gtest', self.run_tests or spec.satisfies('dev_path=*')),
```

**Review by psakievich — CHANGES_REQUESTED** [review]
Looks good! Thanks for the fast turn around time on this @sethrj .  I would ask that the `GTest` stuff also support the `spack develop` workflow.  I find that I rarely use the `run_tests` protocol with standard builds due to CI/CD checking, but I always need to run the tests when doing developer workflow.
Also let me run a couple tests to make sure I can get `nalu-wind` to run with it's own dependency on `gtest`.  A few years ago there was an issue with an attempt to use our own direct dependency of `gtest` conflicting with `trilinos`'s version when that was tried on the project.  If that is still the case then I will probably need to request that `gtest` stays on in `trilinos` until that can get sorted out.

File: var/spack/repos/builtin/packages/trilinos/package.py
Comment (sethrj) [reply var/spack/repos/builtin/packages/trilinos/package.py]: Cool trick! Good to know how to detect `spack develop`.

Comment (sethrj) [reply var/spack/repos/builtin/packages/trilinos/package.py]: > Also let me run a couple tests to make sure I can get `nalu-wind` to run with it's own dependency on `gtest`. A few years ago there was an issue with an attempt to use our own direct dependency of `gtest` conflicting with `trilinos`'s version when that was tried on the project. If that is still the case then I will probably need to request that `gtest` stays on in `trilinos` until that can get sorted out.

I guess we'll have to restore a gtest variant, because I'm working on other projects where having gtest enabled *breaks* downstream projects: vendored dependencies are antithetical to large software ecosystems, so we should see if the tribits-installed gtest can use an external googletest if trilinos' gtest package has to stay enabled.

Comment (psakievich) [reply var/spack/repos/builtin/packages/trilinos/package.py]: > Cool trick! Good to know how to detect `spack develop`.

Yeah I'm thinking we should eventually add a function at the `package` level that just returns the result of this conditional.  something like `uses_tests()`.

**Comment by sethrj** [comment]
@spackbot run pipeline

**Comment by sethrj** [comment]
@psakievich no pressure but is there any update on this?

**Comment by psakievich** [comment]
@sethrj I haven't had a chance to run tests yet, but after thinking about it I'm 90% sure we need to just have `gtest` be a complete variant controlled through the spec.  I believe the issue for `nalu-wind` is that we re-use `stk` unittest fixtures since our data structures come from `stk`.  This means unless `stk` switches to an external `gtest` we will likely need to link to the same version that `stk` uses.  So the current variant will make doing `spack develop` for `nalu-wind` dependent on a develop spec for `trilinos` as well.

**Comment by sethrj** [comment]
@psakievich OK, that makes sense then. I'll restore the variant on this branch.

**Comment by sethrj** [comment]
And added googletest to the list of vendored dependencies in https://github.com/spack/spack/issues/23616 .

**Review by psakievich — APPROVED** [review]

**Comment by kuberry** [comment]
Hi @sethrj, I tested this PR using the configurations relevant to Trilinos 12.12.1 in Xyce and testing +/~ mpi for the newest Trilinos on macOS (and they built). Are there any variants you think should be explicitly tried?

**Comment by sethrj** [comment]
@kuberry Thanks. I think the only new case (`+gtest`) has been tested -- I just need a maintainer approval to merge. (@alalazo  ?)

**Review by alalazo — APPROVED** [review]
:slightly_smiling_face:

**Pull request merged**
